src/ParkChecklist/ParkChecklist.js

```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
+  const [loading, setLoading] = useState(true);
+  const [parks, setParks] = useState([]);
+  const [park_code, setParkCode] = useState(null);
+  const [checkedItems, setCheckedItems] = useState([]);
+  const [selectedState, setSelectedState] = useState("");
+  const [showResults, setShowResults] = useState(false);
+  const [foundPark, setFoundPark] = useState(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const navigate = useNavigate();
+  const parksPerPage = 48;
+  const states = [
+    "AL",
+    "AK",
+    "AZ",
+    "AR",
+    "CA",
+    "CO",
+    "CT",
+    "DE",
+    "FL",
+    "GA",
+    "HI",
+    "ID",
+    "IL",
+    "IN",
+    "IA",
+    "KS",
+    "KY",
+    "LA",
+    "ME",
+    "MD",
+    "MA",
+    "MI",
+    "MN",
+    "MS",
+    "MO",
+    "MT",
+    "NE",
+    "NV",
+    "NH",
+    "NJ",
+    "NM",
+    "NY",
+    "NC",
+    "ND",
+    "OH",
+    "OK",
+    "OR",
+    "PA",
+    "RI",
+    "SC",
+    "SD",
+    "TN",
+    "TX",
+    "UT",
+    "VT",
+    "VA",
+    "WA",
+    "WV",
+    "WI",
+    "WY",
+  ];
 
-    const fetchParkDetails = async (park_code) => {
-        try {
-            const response = await axios.get(`${apiLink}/${park_code}`);
-            const selectedPark = response.data.data;
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(apiLink);
+      setParks(response.data.data);
+      setLoading(false);
+    } catch (error) {
+      console.error("Error fetching data:", error);
+      setLoading(true);
+    }
+  };
 
-            if (selectedPark) {
-                setParkCode(selectedPark.attributes.park_code);
-                setFoundPark(selectedPark);
-            } else {
-                console.error(`Park with id ${park_code} not found.`);
-            }
-            console.log(selectedPark.attributes.park_code)
-            navigate(`/Parks/${selectedState}/${park_code}`);
-        } catch (error) {
-            console.error("Error fetching park details:", error);
-        }
-    };
+  useEffect(() => {
+    fetchData();
+  }, [apiLink, selectedState, showResults, currentPage, navigate]);
 
-    const fetchData = async () => {
-        try {
-            const response = await axios.get(apiLink);
-            setParks(response.data.data);
-            setLoading(false);
-        } catch (error) {
-            console.error("Error fetching data:", error);
-            setLoading(true);
-        }
-    };
+  const handleCheckboxChange = (parkId) => {
+    setCheckedItems((prevCheckedItems) => {
+      if (prevCheckedItems.includes(parkId)) {
+        return prevCheckedItems.filter((id) => id !== parkId);
+      } else {
+        return [...prevCheckedItems, parkId];
+      }
+    });
+  };
 
-    useEffect(() => {
-        fetchData();
-    }, [apiLink, selectedState, showResults, currentPage, navigate]);
+  const fetchParkDetails = async (park_code) => {
+    try {
+      const response = await axios.get(`${apiLink}/${park_code}`);
```

Comment:
i love the use of interpolation and imports to make the fetch work, it makes the code super dynamic